tanjun/abc.py

```diff
@@ -3225,9 +3775,24 @@ async def declare_application_command(
         *,
         application: typing.Optional[hikari.SnowflakeishOr[hikari.PartialApplication]] = None,
         guild: hikari.UndefinedOr[hikari.SnowflakeishOr[hikari.PartialGuild]] = hikari.UNDEFINED,
-    ) -> hikari.Command:
-        """Declare a single slash command for a bot.
+    ) -> hikari.SlashCommand:
+        """Deprecated alias of `Client.declare_slash_command`.
+
+        .. deprecated:: 0.0.1
+            Use `Client.declare_slash_command` instead.
+        """
 
+    @abc.abstractmethod
+    async def declare_slash_command(
```

Comment:
Remove declare_slash_command and switch back to having declare_application_command except use overloads to infer command specific return types